{
  "id": "EUVD-2024-0031",
  "enisaUuid": "60ba1913-8636-3c91-859f-b3a4a7a59dce",
  "description": "cryptography is a package designed to expose cryptographic primitives and recipes to Python developers. Starting in version 38.0.0 and prior to version 42.0.4, if `pkcs12.serialize_key_and_certificates` is called with both a certificate whose public key did not match the provided private key and an `encryption_algorithm` with `hmac_hash` set (via `PrivateFormat.PKCS12.encryption_builder().hmac_hash(...)`, then a NULL pointer dereference would occur, crashing the Python process. This has been resolved in version 42.0.4, the first version in which a `ValueError` is properly raised.",
  "datePublished": "Feb 21, 2024, 7:15:09 AM",
  "dateUpdated": "Feb 6, 2025, 2:34:24 AM",
  "baseScore": 7.5,
  "baseScoreVersion": "3.1",
  "baseScoreVector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
  "references": [
    "https://github.com/pyca/cryptography/security/advisories/GHSA-6vqw-3v5j-54x4",
    "https://github.com/pyca/cryptography/commit/97d231672763cdb5959a3b191e692a362f1b9e55",
    "https://github.com/pyca/cryptography/pull/10423",
    "https://github.com/pyca/cryptography/pull/10423",
    "https://nvd.nist.gov/vuln/detail/CVE-2024-26130"
  ],
  "aliases": [
    "CVE-2024-26130",
    "PYSEC-2024-225"
  ],
  "assigner": "GitHub_M",
  "epss": 0.35,
  "enisaIdProduct": [
    {
      "id": "81362c07-62a4-31bb-a00c-44a57a9a2d02",
      "product": {
        "name": "cryptography"
      }
    },
    {
      "id": "a33cf18e-6524-33c5-afbc-b4819a239b7c",
      "product": {
        "name": "cryptography"
      },
      "product_version": "38.0.0, < 42.0.4"
    }
  ],
  "enisaIdVendor": [
    {
      "id": "e9870140-df08-3dbd-96dd-30ead3abbccc",
      "vendor": {
        "name": "pyca"
      }
    }
  ]
}